pkgs/top-level/all-packages.nix

```diff
@@ -22068,6 +22068,57 @@ with pkgs;
                 );
     }).config.system.build;
 
+
+  /*
+   * Run a NixOS VM network test.
+   *
+   * This is equivalent to `import ./make-test.nix` from the NixOS manual
+   * For details see https://nixos.org/nixos/manual/index.html#sec-nixos-tests
+   *
+   * Parameter:
+   *   A NixOS VM test network, or path to it. Example:
+   *
+   *      { lib, ... }:
+   *      { name = "my-test";
+   *        nodes = {
+   *          machine-1 = someNixOSConfiguration;
+   *          machine-2 = ...;
+   *        }
+   *      }
+   *
+   * Result:
+   *   A derivation that runs the VM test.
+   *
+   * For the interaction between Nixpkgs and NixOS configuration
+   * consult the pkgs.nixos function documentation.
+   */
+  nixosTest =
+    let
+      /* The nixos/lib/testing.nix module, preapplied with arguments that
+       * make sense for this evaluation of Nixpkgs.
+       */
+      nixosTesting =
+        (import ../../nixos/lib/testing.nix {
+          inherit (pkgs.stdenv.hostPlatform) system;
+          inherit pkgs;
+          extraConfigurations = [(
+            { lib, ... }: {
+              config.nixpkgs.pkgs = lib.mkDefault pkgs;
```

Comment:
@aszlig I have found a solution for this in #49256. This function does not introduce import (`NIX_PATH`) impurities; it should actually make it easier to avoid accidental import impurities because it calls the NixOS files with the right arguments that avoid impure defaults.